dockstore-webservice/src/main/java/io/dockstore/webservice/core/Version.java

```diff
@@ -67,6 +69,11 @@
 @Entity
 @ApiModel(value = "Version", description = "Base class for versions of entries in the Dockstore")
 @Inheritance(strategy = InheritanceType.TABLE_PER_CLASS)
+
+@NamedQueries({
+        @NamedQuery(name = "Version.findVersionInEntry", query = "SELECT v FROM Version v WHERE :entryId = v.parent.id AND :versionId = v.id")
```

Comment:
It probably doesn't hurt, but both tagids and entryids are unique, so it seems like the parentid check is redundant.

Reply:
The concern I had was making sure that doing GET /workflows/<id of foo>/versions/<id of version in different workflow> doesn't work. Especially if the user doesn't have permissions to another workflow.

Reply:
I think the checkUser function already accomplishes that.

Reply:
Yeah so checkOptionalAuthRead() checks if the entry is published or the user has access, so it takes care of the permission issue.

Reply:
It only takes care of the permissions issue if the version belongs to the entry that you checked. :) Which is why I think you do need the parent id here. Or maybe we're saying the same thing.

Reply:
Yeah, i think we're saying the same thing. I'm keeping the parentid